CMakeLists.txt

```diff
@@ -234,31 +265,58 @@ if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
   endif()
 endif()
 
+
+message(STATUS "CUDA Compiler: ${CMAKE_CUDA_COMPILER}")
+
 # OpenMP SIMD support
-if(CXX_HAVE_OMP_SIMD)
+if(CXX_HAVE_OMP_SIMD AND NOT DALI_CLANG_ONLY)
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fopenmp-simd")
 else()
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-unknown-pragmas")
 endif()
 
 # Discover what architectures does nvcc support
-CUDA_find_supported_arch_values(CUDA_supported_archs ${CUDA_known_archs})
+if (DALI_CLANG_ONLY)
+  set(USED_CUDA_COMPILER ${CMAKE_CXX_COMPILER})
+else()
+  set(USED_CUDA_COMPILER ${CMAKE_CUDA_COMPILER})
+endif()
+
+
+message(STATUS "CUDA .cu files compiler: ${USED_CUDA_COMPILER}")
+
+CUDA_find_supported_arch_values(CUDA_supported_archs ${USED_CUDA_COMPILER} ${CUDA_known_archs})
```

Comment:
Isn't that duplication of L293?

Reply:
No, It's the CUDA_supported_archs - what compiler supports from the predefined set,
vs the CUDA_targeted_archs - what the user requested. The logic was already there.

Reply:
OK